Blizzard.IO.RabbitMQ/Implementaions/NetqRabbitConsumer.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using Blizzard.IO.Core;
+using Blizzard.IO.RabbitMQ.Entities;
+using EasyNetQ;
+using EasyNetQ.Topology;
+
+namespace Blizzard.IO.RabbitMQ.Implementaions
+{
+    public class NetqRabbitConsumer<TData> : IConsumer<TData>, IConsumerWithMetadata<TData, RabbitMessageProperties>
+    {
+        private readonly IBus _netqBus;
+        private readonly IDeserializer<TData> _deserializer;
+        private readonly IConcreteTypeDeserializer<TData> _concreteTypeDeserializer;
+        private readonly IQueue _sourceQueue;
```

Comment:
Dont use eastnetq's `IQueue` interface, it is a bad interface and the only way to receive an instance of this interface is to call the `DeclareQueue` method in `IAdvanceBus` and we dont want our users to mess with the advance bus and declare like this queues and exchanges.
This the reason the `RabbitQueue` exist

Reply:
The `consume` function needs an `IQueue` instance in order to get data
I don't think that the cunsumer is responsible to start converting `RabbitQueue` into `IQueue`
Moreover, now that `IRabbitMenagmentOps` does not exist anymore, `RabbitQueue` and `RabbitExchange` really there only for `IRabbitConfigurator`, everybody else (like your team) is going to declare topology directly on the `IBus` 
Therefore I believe that the consumer should get `IQueue`, especially because it is already coupled to easynetq by design

Reply:
After further research you can create a `Queue` that implements `IQueue`.
I have no problem having IQueue as member but I think we need to receive `RabbitQueue` from the user.
We cant expect the user to call the declare queue function of the bus.
what if he uses the configurator? he would call the declare queue again just to get this interface?. you can easilly caonvert between the RabbitQueue to EasyNetq's `Queue` in the ctor

Reply:
The question is whether the `IQueue` declared beforehand and the `IQueue` I would create in the consumer  will refer to the same queue in the rabbit server

Reply:
Talk to me about it in person

Reply:
f